torch/_dynamo/variables/nn_module.py

```diff
@@ -212,11 +212,14 @@ def record_nn_module_stack():
                 # for lazy modules, run the pre-hooks which will update the type
                 # TODO mlazos: we don't fully support all of the hooks that exist,
                 # so restrict using __call__ only to lazy modules for now
+                class_source = AttrSource(self.source, "__class__")
```

Comment:
should we assert source here?

Reply:
can do. not sure which assert you meant but i added 2.  probably the self.source one is the more critical but the second one can't hurt